pex/pyenv.py

```diff
@@ -0,0 +1,205 @@
+                active_versions = self.pyenv.active_versions(search_dir=search_dir)
+                if active_versions:
+                    binary_name = os.path.basename(self.path)
+                    if self.name == "python" and not self.major and not self.minor:
+                        for pyenv_version in active_versions:
+                            if pyenv_version[0] in self._PYENV_CPYTHON_VERSION_LEADING_CHARS:
+                                TRACER.log(
+                                    "{} has active version {}".format(self, pyenv_version), V=6
+                                )
+                                return self.pyenv.python(pyenv_version, binary_name=binary_name)
+
+                    prefix = "{name}{major}{minor}".format(
+                        name="" if self.name == "python" else self.name,
+                        major=self.major or "",
+                        minor=".{}".format(self.minor) if self.minor else "",
+                    )
+                    for pyenv_version in active_versions:
+                        if pyenv_version.startswith(prefix):
+                            TRACER.log("{} has active version {}".format(self, pyenv_version), V=6)
+                            return self.pyenv.python(pyenv_version, binary_name=binary_name)
+
+                TRACER.log("{} is not activated.".format(self), V=6)
+                return None
+
+    def as_shim(self, binary):
+        # type: (str) -> Optional[Shim]
+        """View the given binary path as a pyenv shim script if it is one."""
+        return self.Shim.parse(self, binary)
+
+    @staticmethod
+    def _read_pyenv_versions(version_file):
+        # type: (str) -> Iterator[str]
+        with open(version_file) as fp:
+            for line in fp:
+                for version in line.strip().split():
+                    yield version
+
+    @staticmethod
+    def _find_local_version_file(search_dir):
+        # type: (str) -> Optional[str]
+        while True:
+            local_version_file = os.path.join(search_dir, ".python-version")
+            if os.path.exists(local_version_file):
+                return local_version_file
+            parent_dir = os.path.dirname(search_dir)
+            if parent_dir == search_dir:
+                return None
+            search_dir = parent_dir
+
+    def active_versions(self, search_dir=None):
+        # type: (Optional[str]) -> Tuple[str, ...]
+        """Reports the active pyenv versions for the given starting search directory or $PWD."""
+
+        # See: https://github.com/pyenv/pyenv#choosing-the-python-version
+        with TRACER.timed("Finding {} active versions...".format(self), V=6):
+            shell_version = os.environ.get("PYENV_VERSION")
+            if shell_version:
+                TRACER.log(
+                    "Found active pyenv version of PYENV_VERSION={}".format(shell_version), V=6
+                )
+                return (shell_version,)
```

Comment:
The shell version can be multiple versions (separated by a colon), don't we need to split the string here?

```
$ pyenv shell 3.7.3 2.7.15
$ echo $PYENV_VERSION
3.7.3:2.7.15
```

Reply:
And would be good to test this case.

Reply:
Aha. I did not know about that. Thanks! Added handling and a test case.